spec.html

```diff
@@ -19473,17 +19511,103 @@
+
+      <emu-table id="table-script-records" caption="Script Record Fields">
+        <table>
+          <thead>
+          <tr>
+            <th>
+              Field Name
+            </th>
+            <th>
+              Value Type
+            </th>
+            <th>
+              Meaning
+            </th>
+          </tr>
+          </thead>
+          <tbody>
+          <tr>
+            <td>
+              [[Realm]]
+            </td>
+            <td>
+              Realm Record | *undefined*
+            </td>
+            <td>
+              The Realm within which this script was created. *undefined* if not yet assigned.
+            </td>
+          </tr>
+          <tr>
+            <td>
+              [[Environment]]
+            </td>
+            <td>
+              Lexical Environment | *undefined*
+            </td>
+            <td>
+              The Lexical Environment containing the top level bindings for this script. This field is set when the script is instantiated.
+            </td>
+          </tr>
+          <tr>
+            <td>
+              [[ECMAScriptCode]]
+            </td>
+            <td>
+              a parse result
+            </td>
+            <td>
+              The result of parsing the source text of this module using |Script| as the goal symbol.
+            </td>
+          </tr>
+          <tr>
+            <td>
+              [[HostDefined]]
+            </td>
+            <td>
+              Any, default value is *undefined*.
+            </td>
+            <td>
+              Field reserved for use by host environments that need to associate additional information with a script.
+            </td>
+          </tr>
+          </tbody>
+        </table>
+      </emu-table>
+    </emu-clause>
+
+    <emu-clause id="sec-parse-script" aoid="ParseScript">
+      <h1>ParseScript ( _sourceText_, _hostDefined_ )</h1>
+
+      <p>The abstract operation ParseScript with arguments _sourceText_ and _hostDefined_ creates a Script Record based upon the result of parsing _sourceText_ as a |Script|. ParseScript performs the following steps:</p>
+
+      <emu-alg>
+        1. Assert: _sourceText_ is an ECMAScript source text (see clause <emu-xref href="#sec-ecmascript-language-source-code"></emu-xref>).
+        1. Parse _sourceText_ using |Script| as the goal symbol and analyze the parse result for any Early Error conditions. If the parse was successful and no early errors were found, let _body_ be the resulting parse tree. Otherwise, let _body_ be an indication of one or more parsing errors and/or early errors. Parsing and early error detection may be interweaved in an implementation dependent manner. If more than one parse or early error is present, the number and ordering of reported errors is implementation dependent but at least one error must be reported.
+        1. If _body_ is an abrupt completion or error indication, then
+          1. Throw a SyntaxError exception.
```

Comment:
The sentence "If more than one parse or early error is present, the number and ordering of reported errors is implementation dependent but at least one error must be reported." allows to report multiple errors at once, but the next step "Throw a SyntaxError exception." limits this to a single error.

Also "Throw a SyntaxError exception." is not correct for early ReferenceErrors. (*Me shakes fist at early reference errors.*) The current 15.2.1.16.1 ParseModule abstract operation needs to fixed, too.